packages/template-retail-react-app/app/hooks/use-locale.js

```diff
@@ -5,23 +5,13 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import useSite from './use-site'
-import {useMemo} from 'react'
-import {resolveLocaleFromUrl} from '../utils/utils'
-import {useLocation} from 'react-router-dom'
+import {useContext} from 'react'
+import {LocaleContext} from '../contexts'
 
 /**
- * This hook returns the locale object based on current location
- * @return {object} locale
+ * Custom React hook to get the locale
+ * @returns {locale: Object, setLocale: function}
  */
-const useLocale = () => {
-    const {pathname, search} = useLocation()
-    const site = useSite()
-    const locale = useMemo(() => {
-        return resolveLocaleFromUrl(`${pathname}${search}`)
-    }, [pathname, search, site])
-
-    return locale
-}
+const useLocale = () => useContext(LocaleContext)
```

Comment:
If possible, let's reuse what `react-intl` has given us. It has `useIntl` hook, for example. Calling it would give us an Intl object, which looks like this: https://formatjs.io/docs/react-intl/api/#intlshape

Reply:
It looks like this was already addressed in this comment:  https://github.com/SalesforceCommerceCloud/pwa-kit/pull/667#discussion_r928003090